File: problem35/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int goodNodes(TreeNode root) {
        if (root == null)
            return 0;
        return 1 + goodLeaves(root.val, root.left) + goodLeaves(root.val, root.right);
    }
    private int goodLeaves(int max, TreeNode root) {
        if (root == null)
            return 0;
        if (root.val >= max)
            return 1 + goodLeaves(root.val, root.left) + goodLeaves(root.val, root.right);
        else
            return 0 + goodLeaves(max, root.left) + goodLeaves(max, root.right);
    }
}

File: problem35/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(big, root):
            if not root: return 0
            isGood = 0
            if root.val >= big:
                isGood = 1
                big = root.val
            return isGood + dfs(big, root.left) + dfs(big, root.right)
        return dfs(-10000, root)
